docs/design/nab_and_nar_status_update.md

```diff
@@ -0,0 +1,192 @@
+# Controller Workflow: NonAdminBackup and NonAdminRestore Status Update
+
+## Overview
+
+This document outlines the design around updating NonAdminBackup (NAB) and NonAdminRestore (NAR) objects' Statuses.
+
+## NonAdminBackup and NonAdminRestore Status
+
+The `status` field of NAB and NAR objects contains the following fields:
+- `phase`
+- `conditions`
+- `veleroBackup` for NAB and `veleroRestore` for NAR, which contains name, namespace and status of the related Velero object.
+
+which are updated by NAB and NAR controllers.
+
+Only one update call should be performed per reconcile of NAB and NAR objects. Controller can requeue or depend on predicates if more updates are needed in that object.
+
+### Phase
+
+The `phase` field is a simple one high-level summary of the lifecycle of the objects, that only moves forward. Once a `phase` changes, it can not return to the previous value.
+
+It is always a one well defined value, that is intended to be a comprehensive state of a NAB or NAR object.
+
+Those are are the possible values for phase:
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| New | *NonAdminBackup/NonAdminRestore* resource was accepted by the NAB/NAR Controller, but it has not yet been validated by the NAB/NAR Controller |
+| BackingOff | *NonAdminBackup/NonAdminRestore* resource was invalidated by the NAB/NAR Controller, due to invalid Spec. NAB/NAR Controller will not reconcile the object further, until user updates it |
+| Created | *NonAdminBackup/NonAdminRestore* resource was validated by the NAB/NAR Controller and Velero *Backup/restore* was created. The Phase will not have additional information about the *Backup/Restore* run |
```

Comment:
There was discussion (don't remember with whom) if we should add 4th and 5th Phase: `Completed` when the Velero operation completes with success and `Failed` when the backup failed. I know it's a bit of duplication with Condition, but the phase is more important for the user.

Reply:
I think we will, in the end, add more conditions. But when we do, we update. Otherwise I am trying to predict the future

Reply:
Completed and Failed are the two conditions straight from the Velero `Backup` workflow, so it's not prediction but based on actual `Backup` current behavior.